ext/standard/tests/general_functions/parse_ini_basic.data

```diff
@@ -141,3 +143,11 @@ double_quoted_multiline = "Lorem \"ipsum\"""
  dolor"
 dollar_test = "\${test}"
 unescaped ="\n\r\t"
+
+[variable-fallback]
+hello = "Hello, ${undefined:-world}!"
+simple = ${hi_is_this_undefined:-no, this is fallback!}
+hostname = localhost:${undefined_port:-8080}
+quoted = "foo:${undefined:-bar}"
+empty1 = ${undefined:-}
+empty2 = ${undefined}
```

Comment:
Are these enough? Should I write some more test cases?

Reply:
I'm missing some tests where the variable is not undefined. You should also add tests for falsy values like `null`, `false`, `''` etc. for the desired semantics. You might need `INI_SCANNER_TYPED` to avoid values from being coerced to strings right away.